src/library_fs.js

```diff
@@ -421,15 +421,64 @@ mergeInto(LibraryManager.library, {
           },
           isAppend: {
             get: function() { return (this.flags & {{{ cDefine('O_APPEND') }}}); }
-          }
+          },
+          flags: {
+            get: function() { return this.shared.flags; },
+            set: function(value) { this.shared.flags = value; }
+          },
+          position: {
+            get: function() { return this.shared.position; },
+            set: function(value) { this.shared.position = value; }
+          },
+#if NODERAWFS
+          /* refcount is a stream attribute only used by NODERAWFS
+           *
+           * Duplicated streams share flags and seek position, as described in dup(2):
+           *
+           *   http://man7.org/linux/man-pages/man2/dup.2.html
+           *
+           * In-memory streams simply use a shared object to accomplish this, so that
+           * updating either property on one fd will change it on duplicated fds as well.
+           *
+           * Node, however, creates a node fd upon opening a file, and closing
+           * the node fd will also break duplicated fds. To work around this,
+           * we increment the refcount for the fd upon duplicating the stream, and
+           * only close the fd when all streams referencing the fd are closed.
+           */
+          refcount: {
+            get: function() { return this.shared.refcount; },
+            set: function(value) { this.shared.refcount = value; }
+	  }
+#endif
         });
       }
-      // clone it, so we can return an instance of FSStream
+
       var newStream = new FS.FSStream();
       for (var p in stream) {
-        newStream[p] = stream[p];
+        if (sharedProperties.indexOf(p) >= 0) {
+          if (!("shared" in newStream)) {
```

Comment:
I think these lines might be related to the closure issue.

When closure minifies property names, there is no `stream.shared`, instead there will be `stream.a` or something like that. So it's not valid to check using the string name, `stream['shared']`. Instead, you can check for `stream.shared`, which closure will minify.

I think that also means that `sharedProperties` is not going to work, as it contains string names. You probably need to check for `stream.X` directly for each of those properties.

Reply:
Yeah, that was it. That code was a little ugly anyway. Because of the accessor functions defined earlier, it's not necessary to separate out the shared properties, and I've fixed that.